OpenSim/Simulation/Model/Point.cpp

```diff
@@ -0,0 +1,129 @@
+ * Author(s): Ajay Seth                                                       *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+//=============================================================================
+// INCLUDES
+//=============================================================================
+#include "Point.h"
+
+//=============================================================================
+// STATICS
+//=============================================================================
+using namespace std;
+using namespace OpenSim;
+using SimTK::Mat33;
+using SimTK::Vec3;
+
+//=============================================================================
+// CONSTRUCTOR(S)
+//=============================================================================
+//_____________________________________________________________________________
+/**
+ * Default constructor.
+ */
+Point::Point() : ModelComponent()
+{
+    setAuthors("Ajay Seth");
+}
+
+
+void Point::extendAddToSystem(SimTK::MultibodySystem& system) const
+{
+    Super::extendAddToSystem(system);
+    SimTK::Vec3 v(SimTK::NaN);
+    // If the properties, topology or coordinate values change, 
+    // Stage::Position will be invalid.
+    addCacheVariable("location", v, SimTK::Stage::Position);
+    addCacheVariable("velocity", v, SimTK::Stage::Velocity);
+    addCacheVariable("acceleration", v, SimTK::Stage::Acceleration);
+}
+
+const SimTK::Vec3& Point::getLocationInGround(const SimTK::State& s) const
+{
+    if (!getSystem().getDefaultSubsystem().
+        isCacheValueRealized(s, _locationIndex)){
+        //cache is not valid so calculate the transform
+        SimTK::Value<SimTK::Vec3>::downcast(
+            getSystem().getDefaultSubsystem().
+            updCacheEntry(s, _locationIndex)).upd()
+                = calcLocationInGround(s);
+        // mark cache as up-to-date
+        getSystem().getDefaultSubsystem().
+            markCacheValueRealized(s, _locationIndex);
+    }
+    return SimTK::Value<SimTK::Vec3>::downcast(
+        getSystem().getDefaultSubsystem().
+        getCacheEntry(s, _locationIndex)).get();
+}
```

Comment:
Is this really the way that cache variables are to be handled? It is very verbose. Why not use `Component::isCacheVariableValid()` and `Component::markCacheVariableAsValid()`? If your concern is performance, then the interface for cache variables in Component should be updated as a separate issue, and this code should still just use the Component's current interface.